File: gcd/java/gcd.java
public class GCD {
    public static void main(String[] args) {
        int n1 = 81, n2 = -153;
        // Always set to positive
        n1 = ( n1 > 0) ? n1 : -n1;
        n2 = ( n2 > 0) ? n2 : -n2;
        while(n1 != n2)
        {
            if(n1 > n2)
                n1 -= n2;
            else
                n2 -= n1;
        }
        System.out.println("G.C.D = " + n1);
    }
}

File: gcd/gcd.pas
program gcd (input, output);
{gives out the greatest common divisor of two integers}

	var
	a,
	b : integer;
	
begin
	writeln ('Please type in two integers > 0. ');
	readln(a);
	readln(b);
	if (a <= 0) or (b <= 0) then
		writeln ('Error!')
	else
	begin
		write ('The gcd of ', a, ' and ', b, ' is ');
		
		while a <> b do
			if a > b then
				a := a - b
			else
				b := b - a;
		writeln (a, '.')
	end
end. 
